code/modules/reagents/chemistry/reagents/blob_reagents.dm

```diff
@@ -135,17 +136,44 @@
 	if(iscarbon(M))
 		M.emote("scream")
 
+/datum/reagent/blob/regenerative_materia
+	name = "Regenerative Materia"
+	id = "regenerative_materia"
+	description = "will do low toxin damage and cause targets to believe they are fully healed."
+	color = "#C8A5DC"
+	message_living = ", and you feel <i>alive</i>"
+
+/datum/reagent/blob/regenerative_materia/reaction_mob(mob/living/M, method=TOUCH, reac_volume, show_message, touch_protection, mob/camera/blob/O)
+	reac_volume = ..()
+	if(M.reagents)
+		M.reagents.add_reagent("regenerative_materia", 0.2*reac_volume)
+	if(M)
```

Comment:
There's a few of these if(M) checks after you already accessed M variables. Either remove them where there's no chance of mob getting deleted or move them to the top of the block.